Revised button guidelines for Plasma (#2425)
coveo/plasma: packages/website/docs/Button.md

## Best Practices

Button label should allow users to **foresee what will happen** when clicking them. 

Avoid multiplying buttons within a page. A page should have **only one primary action button**. If several secondary action buttons are required, consider using [Actionable items](#/form/ActionableItem) instead.

When using button groups, such as for "Save" and "Cancel" actions, always put the action that move the user forward through his journey **to the right**. For instance, order buttons as "Cancel" then "Save".

## Labeling

Keep labels short, preferably **under three words**.

> Use **trigger words** to clearly indicate the action performed by the button.

Reviewer: what is a trigger word? An action? :thinking:

Author: Trigger words are the verbs/action words that will influence a user into clicking. 
They are more specific than regular verbs. Maybe we could call them "Commands" verbs 
@jlevesquecoveo any thoughts on this?

E.g.: 
Using "View Profile" instead of "Open Profile", 
Using "Publish Story" instead of  "Save Story" (depending on the context)
Using "Create X" instead of  "Save" at the end of a stepper modal when we add something in the platform.

## Variants

| **Type** | **Purpose** |
| **Primary** | Use primary buttons for the main action you want to draw the user attention to. There should be only one primary action in a section or page. |
| **Secondary** | Use secondary buttons to display action of lesser importance or to display multiple actions of similar importance. |
| **Left icon + label** | Add a left icon to add quick and visual clarity to what the action is or draw attention to the button. Make sure the icon is highly recognizable. |
| **Left icon + label** | Use the right icon to indicate that additional options are available upon selecting the button. This means that the action will not be carried right away upon clicking the button, but a menu overlay would be displayed where the user can select the exact action to be performed. For instance, when the action is "Create" and options are to create from a template or a blank file. |
| **Right icon + label** | Use only an icon when space is an issue or when the icon is highly and instantly recognizable. For instance, the "Setting" button is commonly presented only using a cog icon. Never use icon only button for primary actions. |

> | **Append and prepend separator** | Adding a separator is an aesthetic choice especially used when two or more buttons are stacked vertically. Use a separator to align the icon to the right or left or the button. Note that all buttons must have the same width when vertically stacked. |

Reviewer: All this section is really hard to read, we're you attempting to make a table? :eyes: 
![image](https://user-images.githubusercontent.com/63734941/152883150-e309f688-c347-4129-91c8-43a23c8abf4a.png)

Reviewer 2: Here's the syntax for a table, you might be just missing the line `| ---- | --- | ---- |` to delimit the table header

![image](https://user-images.githubusercontent.com/35579930/153032628-3e178f49-49b5-4829-b84c-ac04db59b863.png)

Author: I fix it. let me know if it's better

Author: fixed it for real (I think)

## Best Practices

Button label should allow users to **foresee what will happen** when clicking them. 

Avoid multiplying buttons within a page. A page should have **only one primary action button**. If several secondary action buttons are required, consider using [Actionable items](#/form/ActionableItem) instead.

When using button groups, such as for "Save" and "Cancel" actions, always put the action that move the user forward through his journey **to the right**. For instance, order buttons as "Cancel" then "Save".

## Labeling

Keep labels short, preferably **under three words**.
Use **trigger words** to clearly indicate the action performed by the button.

> Write labels so **the first word is a verb**. Use noun without article, to specify the context when required. For example, write "Add filter" rather than "Add a filter" but only write "Save" and "Cancel".

Reviewer 3: ```suggestion
The first word of the label should be a verb. If adding a noun is required for context, avoid including an article. For example, write "Add filter" rather than "Add a filter", but write "Save" and "Cancel".
```

> When using button to confirm an action that will be executed, always use an active word to **clearly state the resulting action**. For instance, if user must confirm the deletion of a file, write "Delete" and "Cancel" rather than "Yes" and "No".

Reviewer 3: ```suggestion
When using a button to have users confirm the action to execute, use an active word to **clearly state the resulting action**. For instance, if users must confirm the deletion of a file, write "Delete" and "Cancel" rather than "Yes" and "No".
```

## Variants

| **Type**      | **Purpose** |
| --------- | --------- |

> | **Primary** | Use primary buttons for the main action you want to draw the user attention to. There should be only one primary action in a section or page. |

Reviewer 3: ```suggestion
| **Primary** | A primary button draws the users' attention to the main action. There should be only one primary button per section or page. |
```

> | **Primary** | Use primary buttons for the main action you want to draw the user attention to. There should be only one primary action in a section or page. |

Reviewer 3: I don't think we need to put **Primary** in bold since it appears in a different column. It's already kind of highlighted.

> | **Secondary** | Use secondary buttons to display action of lesser importance or to display multiple actions of similar importance. |

Reviewer 3: ```suggestion
| **Secondary** | A secondary button triggers an action of lesser importance. Multiple secondary buttons may trigger actions of similar importance. |
```

| **Left icon + label** | Add a left icon to add quick and visual clarity to what the action is or draw attention to the button. Make sure the icon is highly recognizable. |

> | **Right icon + label** | Use the right icon to indicate that additional options are available upon selecting the button. This means that the action will not be carried right away upon clicking the button, but a menu overlay would be displayed where the user can select the exact action to be performed. For instance, when the action is "Create" and options are to create from a template or a blank file. |

Reviewer 3: ```suggestion
| **Label + icon** | Add an icon right of the label to indicate that additional options are available when clicking the button. This indicates that clicking the button doesn't immediately trigger the action. Instead, a menu overlay is displayed, allowing users to select the exact action to be performed. For example, use a label and an icon on the right when the action is "Create" and options are to create from a template or a blank file. |
```

> | **Icon only** | Use only an icon when space is an issue or when the icon is highly and instantly recognizable. For instance, the "Setting" button is commonly presented only using a cog icon. Never use icon only button for primary actions. |

Reviewer 3: ```suggestion
| **Icon only** | When space is an issue or when the icon is highly and instantly recognizable, an icon alone may suffice. For instance, the "Setting" button is commonly presented only using a cog icon. However, you should never use an icon-only button for primary actions. |
```

> | **Append and prepend separator** | Adding a separator is an aesthetic choice especially used when two or more buttons are stacked vertically. Use a separator to align the icon to the right or left or the button. Note that all buttons must have the same width when vertically stacked. |

Reviewer 3: ```suggestion
| **Append and prepend separator** | Adding a separator is an aesthetic choice typically made when multiple buttons are stacked vertically. Use a separator to align the icon to the right or left of the button. All buttons must have the same width when vertically stacked. |
```

## Feedback and Validation

> Add a tooltip when the label is hidden to display the name of the related action.

Reviewer 3: ```suggestion
When the label is hidden, add a tooltip displaying the action.
```

---

## Related Components

If your use case doesn't match the guidelines above, consider using one of the following components instead:

> -   [Link](#/foundations/Links) when you need to redirect a user to another section or website

Reviewer 3: ```suggestion
-   [Link](#/foundations/Links) to redirect users to another section or website.
```

> -   [Actionable items](#/form/ActionableItem) when you want to regroup several less important action together.

Reviewer 3: ```suggestion
-   [Actionable items](#/form/ActionableItem) to group several less important action together.
```
